internal/aws/ecsutil/metadata_provider.go

```diff
@@ -0,0 +1,99 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecsutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.uber.org/zap"
+)
+
+type MetadataProvider interface {
+	FetchTaskMetadata() (*TaskMetadata, error)
+	FetchContainerMetadata() (*ContainerMetadata, error)
+}
+
+type ECSMetadataProviderImpl struct {
```

Comment:
This does not need to exported since there is a function to return a Provider and the zero value would cause a panic.

Reply:
Thanks, updated.